### ISSUE TITLE
[BUG] Create rule that already exists doesn't provide ruleid back

### ISSUE THREAD

**Issue by jupiler** [Checkmk/ansible-collection-checkmk.general#663]
**Provide your feedback**
Create rule that already exists doesn't provide ruleid back.
It would be nice if the ruleid would be returned so it can be used in the future as a reference for the rules to be manipulated.

**Component Name**
Component Name: general.rule

**Ansible Version**

```
ansible-playbook [core 2.17.4]
  config file = ~/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/user/.venv/lib/python3.12/site-packages/ansible
  ansible collection location = /opt/user/.venv/collections
  executable location = /opt/user/.venv/bin/ansible-playbook
  python version = 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] (/opt/user/.venv/bin/python)
  jinja version = 3.1.4
  libyaml = True
```

**Collection Version**
```
collections:
  # https://github.com/Checkmk/ansible-collection-checkmk.general/tree/main
  - name: checkmk.general
    version: 5.2.1
```
**Environment**
Checkmk version 2.2.0p27
local os ubuntu 22.04

**Additional context**
The tasks in question.
```
- name: State_present | Create rule in {{ checkmk_rule_ruleset }}
  checkmk.general.rule:
    server_url: "{{ checkmk_rule_host }}"
    site: "{{ checkmk_rule_site }}"
    automation_user: "{{ checkmk_rule_user }}"
    automation_secret: "{{ checkmk_rule_pass }}"
    ruleset: "{{ checkmk_rule_ruleset }}"
    rule:
      rule_id: "{{ checkmk[rule_id_filename].ruleid |default('') }}"
      conditions:
        host_labels: "{{ rule.host_labels | default([]) }}"
        host_name:
          match_on: "{{ rule.host_names }}"
          operator: "{{ rule.host_labels | default('one_of') }}"
        host_tags: "{{ rule.host_tags | default([]) }}"
        service_labels: "{{ rule.service_labels | default([]) }}"
      properties:
        comment: Created with Ansible pipeline
        description: "{{ rule.host_names[0] }} - {{ rule.description | default('no description specified')}}"
        disabled: "{{ rule.disabled | default('False') | bool}}"
      value_raw: "{{ rule.value_raw | string }}"
      location:
        folder: "{{ rule.folder | default(checkmk_rule_folder) }}"
        position: "{{ rule.host_labels | default('top') }}"
    state: present
    validate_certs: "{{ checkmk_validate_cert }}"
  notify: Activate rules
  register: ruleid_info

- name: State_present | Debug show output ruleid_info
  debug:
    var: ruleid_info
  when: ansible_debug_enabled
```
Output of the tasks
```
ok: [checkmk] => {
    "changed": false,
    "content": {},
    "etag": "",
    "http_code": 0,
    "invocation": {
        "module_args": {
            "automation_secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "automation_user": "XXXXXXXXX",
            "rule": {
                "conditions": {
                    "host_labels": [],
                    "host_name": {
                        "match_on": [
                            "XXXXXX"
                        ],
                        "operator": "one_of"
                    },
                    "host_tags": [],
                    "service_labels": []
                },
                "location": {
                    "folder": "/postgresql/",
                    "neighbour": null,
                    "position": "top"
                },
                "properties": {
                    "comment": "Created with Ansible pipeline",
                    "description": "XXXXXX"
                    "disabled": false,
                    "documentation_url": "https://github.com/Checkmk/ansible-collection-checkmk.general/blob/main/plugins/modules/rules.py"
                },
                "rule_id": "",
                "value_raw": "XXXXXXXXX"
            },
            "ruleset": "agent_config:mk_postgres",
            "server_url": "http://XXXXXXXXX",
            "site": "XXXXXXXXX",
            "state": "present",
            "validate_certs": false
        }
    },
    "msg": "Rule already exists with the desired parameters."
}

TASK [checkmk_rule : State_present | Debug show output ruleid_info] ****************************************
task path: 
ok: [checkmk] => {
    "ruleid_info": {
        "changed": false,
        "content": {},
        "etag": "",
        "failed": false,
        "http_code": 0,
        "msg": "Rule already exists with the desired parameters."
    }
}
```

**Comment by robin-checkmk** [comment]
Thanks for reporting this!
At first sight, this looks like a bug to me.
Can you verify, if the same behaviour can be observed on a 2.3.0 site @jupiler?

**Comment by jupiler** [comment]
It seems I got the same behavior with a 2.3.0p11 site.
```
            "site": "m2_3",
            "state": "present",
            "validate_certs": false
        }
    },
    "msg": "Rule already exists with the desired parameters."
}

TASK [checkmk_rule : State_present | Debug show output ruleid_info] ****************************************
task path: 
ok: [checkmk] => {
    "ruleid_info": {
        "changed": false,
        "content": {},
        "etag": "",
        "failed": false,
        "http_code": 0,
        "msg": "Rule already exists with the desired parameters."
    }
}
```

**Comment by jupiler** [comment]
Just patched my local for a test and that seemed to do the trick.

**Comment by robin-checkmk** [comment]
> I just fixed that that in folder.py line #282 removing .lower from str(p.parent):
> 
> ```
>     def _normalize_path(self, path):
>         p = Path(path)
>         if not p.is_absolute():
>             p = Path("/").joinpath(p)
>         return str(p.parent), p.name
> ```

@mike1098 Am I lost, or are you? I think your comment refers to [this issue](https://github.com/Checkmk/ansible-collection-checkmk.general/issues/662), right?

**Comment by robin-checkmk** [comment]
The fix has just been released with [5.3.0](https://github.com/Checkmk/ansible-collection-checkmk.general/releases/tag/v5.3.0).
